### ISSUE TITLE
Mantl 1.2 : ansible installing dnsmasq fails because of missing dependency of network manager 1.0.6

### ISSUE THREAD

**Issue by suyambuganesh** [mantl/mantl#1859]
Hi, Error while using Mantl 1.2 : ansible installing dnsmasq fails because of missing dependency of network manager 1.0.6

- Logs from the failing component
I ran - `ansible-playbook -e @security.yml mantl.yml`

    - `TASK: [dnsmasq | install mantl-dns] *******************************************
failed: [mantl-worker-003] => {"changed": true, "rc": 1, "results": ["Loaded plugins: fastestmirror\nLoading mirror speeds from cached hostfile\n * base: mirror.san.fastserv.com\n * epel: mirrors.kernel.org\n * extras: centos.mirror.ndchost.com\n * updates: mirror.spro.net\nResolving Dependencies\n--> Running transaction check\n---> Package mantl-dns.x86_64 0:1.1.0-5.centos will be installed\n--> Processing Dependency: NetworkManager = 1:1.0.6 for package: mantl-dns-1.1.0-5.centos.x86_64\n--> Processing Dependency: bind-utils = 32:9.9.4 for package: mantl-dns-1.1.0-5.centos.x86_64\n--> Running transaction check\n---> Package bind-utils.x86_64 32:9.9.4-38.el7_3 will be installed\n--> Processing Dependency: bind-libs = 32:9.9.4-38.el7_3 for package: 32:bind-utils-9.9.4-38.el7_3.x86_64\n--> Processing Dependency: liblwres.so.90()(64bit) for package: 32:bind-utils-9.9.4-38.el7_3.x86_64\n--> Processing Dependency: libisccfg.so.90()(64bit) for package: 32:bind-utils-9.9.4-38.el7_3.x86_64\n--> Processing Dependency: libisccc.so.90()(64bit) for package: 32:bind-utils-9.9.4-38.el7_3.x86_64\n--> Processing Dependency: libisc.so.95()(64bit) for package: 32:bind-utils-9.9.4-38.el7_3.x86_64\n--> Processing Dependency: libdns.so.100()(64bit) for package: 32:bind-utils-9.9.4-38.el7_3.x86_64\n--> Processing Dependency: libbind9.so.90()(64bit) for package: 32:bind-utils-9.9.4-38.el7_3.x86_64\n---> Package mantl-dns.x86_64 0:1.1.0-5.centos will be installed\n--> Processing Dependency: NetworkManager = 1:1.0.6 for package: mantl-dns-1.1.0-5.centos.x86_64\n--> Running transaction check\n---> Package bind-libs.x86_64 32:9.9.4-38.el7_3 will be installed\n---> Package mantl-dns.x86_64 0:1.1.0-5.centos will be installed\n--> Processing Dependency: NetworkManager = 1:1.0.6 for package: mantl-dns-1.1.0-5.centos.x86_64\n--> Finished Dependency Resolution\n You could try using --skip-broken to work around the problem\n You could try running: rpm -Va --nofiles --nodigest\n"]}
msg: Error: Package: mantl-dns-1.1.0-5.centos.x86_64 (asteris-mantl-rpm)
           Requires: NetworkManager = 1:1.0.6
           Available: 1:NetworkManager-1.4.0-12.el7.x86_64 (base)
               NetworkManager = 1:1.4.0-12.el7
           Available: 1:NetworkManager-1.4.0-13.el7_3.x86_64 (updates)
               NetworkManager = 1:1.4.0-13.el7_3`
    
- Ansible version (`1.9.6`):
- Python version (`2.7.13`):
- Git commit hash or branch: `tag 1.2`
- Cloud Environment: `AWS`
- Terraform version (`0.8.1`):

**Comment by bbfoto** [comment]
the yum update play updates a few updates beyond NetworkManager 1.0.6.
downgrading was not an option.

I was able to hack around this error.
(and by hack I mean I'm not that concerned about the dependency error with the test environment I'm deploying into...).

I manually forced the install of mantl-dns by doing
ansible all -m shell -a "yumdownloader mantl-dns --destdir /var/tmp"
...
ansible all -m shell -a "sudo rpm -Uhv  --nodeps /var/tmp/mantl-dns-1.1.0-5.centos.x86_64.rpm"
...
then re-ran the ansible-playbook.

I also had to change ./roles/vault/defaults/main.yml vault_package to
vault_package: vault-0.6.0
then reran the playbook once more, and got a clean playbook result.
will see over the next few days if it's all still functional.
i'll post back here if I see anything obviously wrong.

**Comment by suyambuganesh** [comment]
with the above work around, I was able to install and run mantl.

**Comment by distributorofpain** [comment]
Glad someone else found this and posted a fix before today, i had to rebuild the environment and it errored on me as well :(

**Comment by distributorofpain** [comment]
ok, i lied, i keep getting a new error on all three control/master nodes, then the whole thing stops.  I did load the mantl-dns package before starting though.

```
TASK: [kubernetes-master | create/update skydns replication controller] *******
failed: [mantl-do-nyc2-mcontrol-01] => {"failed": true}
msg: error running kubectl (/bin/kubectl --server=http://localhost:8085/ --namespace=kube-system create --filename=/etc/kubernetes/manifests/skydns-rc.yaml) command (rc=1): Error from server: error when creating "/etc/kubernetes/manifests/skydns-rc.yaml": Unexpected response code: 500


FATAL: all hosts have already failed -- aborting
```

**Comment by distributorofpain** [comment]
Well, if i reinstall, it does finish.  But consul is full of failing services.  The interesting part is that the errors all talk about python-pip not being installed, but when try to install python-pip, the servers all report back that it is.  So i am guessing its looking for a specific version?

```
time="2017-01-09T01:08:51Z" level=info msg="Report from checklist" checklist="common health check" report="↴\nTotal: 10\nPassed: 9\nFailed: 1\nOther: 0\nPackage was not found:\n\tPackage name: python-pip\n\tPackage manager: rpm\n\tCommand output: package python-pip is not installed\n" 
```

**Comment by stevendborrelli** [comment]
It's possible that newer versions of centos have changed the package name to `python2-pip`.

**Comment by distributorofpain** [comment]
lol, just as that came into view, i noticed that...  argh

**Comment by distributorofpain** [comment]
So i guess the question would be, whats the best way to fix this?

**Comment by stevendborrelli** [comment]
Do you want to try changing this line? https://github.com/mantl/mantl/blob/3d1db1d78ef1d487f45102d2424ce62a6e934855/roles/common/tasks/main.yml#L109

**Comment by distributorofpain** [comment]
i am not sure that would work.  If you run "yum install python-pip" it does install python-pip, but the package name is python2-pip.  Do the healthchecks look at what was supposed to be installed and look for that or are they hard set to look for python-pip?

I dont mind re-running the install or starting fresh.  My setup is in DO and i am using terraform.  I am however also fighting the mantl-dns issue listed above.  I was forced to download the package to my server and deploy it from there because the servers refuse to install it from yum.  I am using the centos-7-x64 image from DO to install on.  They also have a centos-7-0-x64 (and a 5.5 and 6.5 version ).  The package updates playlist of course puts me right back to using centos-7-x64 though and i am forced to again install mantl-dns from my server.

**Comment by distributorofpain** [comment]
Well i manually installed bind-utils 
```
ansible all -m shell -a "yum install bind-utils -y"
```
and that error went away...

**Comment by distributorofpain** [comment]
ok, so i made the change to the file and reran the install playbook against the running system with the existing error and the health check is still failing with the same message.

**Comment by distributorofpain** [comment]
ok, so i went to the host and edited this file:  "/etc/distributive.d/distributive-common.json" 

Updated the check for python-pip to python2-pip.

That brought the server to green.

**Comment by stevendborrelli** [comment]
@distributorofpain is it still failing with `bind-utils` missing, even though it is present?

**Comment by distributorofpain** [comment]
@stevendborrelli bind-utils wasnt installed apparently.  When i ran the installation command, i didnt get any feedback stating that it was installed already.

**Comment by distributorofpain** [comment]
Playbook to fix it:

```
- hosts: all
  gather_facts: no
  tasks:
  - name: update json healthchecks
    replace: dest=/etc/distributive.d/distributive-common.json regexp='python-pip' replace='python2-pip'
```

I am going to start fresh though, to see if that solves the issue as well.

**Comment by distributorofpain** [comment]
Updating the file before a fresh load didnt fix it.  So installing in a Digital Ocean environment using the centos-7-x64 image ( fresh load ) with terraform.  I had to do the following before running the install script ( and after terraform finished ):

Run these commands:
```
ansible all -m shell -a "yumdownloader mantl-dns --destdir /var/tmp"
ansible all -m shell -a "sudo rpm -Uhv --nodeps /var/tmp/mantl-dns-1.1.0-5.centos.x86_64.rpm"
ansible all -m shell -a "yum install bind-utils -y"
```
Save and run the playbook:
```
- hosts: all
  gather_facts: no
  tasks:
  - name: update json healthchecks
    replace: dest=/etc/distributive.d/distributive-common.json regexp='python-pip' replace='python2-pip'
```

You only need bind-utils on the edge and control nodes according to the consul errors, but i put it on all servers.  Also, if you cannot get the servers to download the package using yum downloader like me, you can download it to a server and pull from a separate server like so(then run the last two commands above):
```
ansible all -m shell -a "wget http://yourserverip/files/mantl-dns-1.1.0-5.centos.x86_64.rpm -P /var/tmp/"
```

ugh dependencies!

**Comment by thomasvincent** [comment]
Was fixed in https://github.com/mantl/mantl/commit/0f2194a494f00dc624042279aa98b89a97a5bd02

**Comment by Theaxiom** [comment]
Interesting, that commit was way before this issue was opened.

**Comment by Cryptophobia** [comment]
@Theaxiom @thomasvincent : Not sure if I follow all of the conversation above but kind of skimmed through pretty fast. Sounds like dnsmasq playbook is not applied to control and edge nodes but the dependencies `python2-pip , python-pip , bind-utils` are needed there.

**Comment by Theaxiom** [comment]
Is this something that is required for 2.0? Or can we move it to 2.1?

**Comment by thomasvincent** [comment]
Required for 2.0

**Comment by Theaxiom** [comment]
We should add those packages to common role

**Comment by Cryptophobia** [comment]
@Theaxiom Yes, we should add these dependencies before the Ansible upgrade

**Comment by Theaxiom** [comment]
Currently testing these packages right now.

**Comment by Theaxiom** [comment]
I have added those packages to the `common` role, however `common Distributive Mantl Health Checks` in `consul` is still reporting: `Package was not found:\n\tPackage name: python-pip\n\tPackage manager: rpm\n\tCommand output: package python-pip is not installed\n" `

**Comment by Theaxiom** [comment]
I think I've identified the problem in #1922 and I am testing the fix right now.

**Comment by Theaxiom** [comment]
Still working on this, you can watch #1922 for updates.

**Comment by Theaxiom** [comment]
The status checks are now green, I added a temporary work-around in https://github.com/mantl/mantl/commit/f19d567faee19c1399ad4fafac4830b235d241b1

**Comment by Theaxiom** [comment]
Alright everyone, I will work on building new RPM packages today and that should get this issue resolved.

**Comment by Theaxiom** [comment]
Here's the related issue for building RPM packages: #1928

**Comment by Theaxiom** [comment]
mantl-dns 1.2 has been released, if anyone wants to test it, here's the repo you have to add:

```
[bintraybintray-mantl-rpm-packages]
name=bintray-mantl-rpm-packages
baseurl=https://dl.bintray.com/mantl/rpm-packages
gpgcheck=0
repo_gpgcheck=0
enabled=1
```

**Comment by KaGeN101** [comment]
Patch this in and it builds now

**Comment by Theaxiom** [comment]
This issue is confirmed resolved and fix is merged into master.
